src/sagemaker/estimator.py

```diff
@@ -1713,6 +1713,7 @@ def _stage_user_code_in_s3(self):
             directory=self.source_dir,
             dependencies=self.dependencies,
             kms_key=kms_key,
+            s3_resource=self.sagemaker_session.s3_resource,
```

Comment:
I'm still unconvinced of this. what if someone passes in both a `sagemaker_session` and an `s3_resource`? I think it would be better to let the user just pass in a `sagemaker_session` with whatever configuration (i.e. `s3_resource`) they want already defined.

Reply:
This function call is wierd; one passes in a boto session (and then we added the s3_resource); we could pass in a sagemaker session instead, and then get out both the boto session and the s3_resource. It would break the API call though.

Reply:
ah, I see. my bad, I was thinking this was in a different part of the code. this is fine, then.